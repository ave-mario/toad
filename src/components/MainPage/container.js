import { connect } from 'react-redux';
import authActions from 'actions/auth.actions';
import MainPage from './view';

const { Creators } = authActions;

const mapStateToProps = state => ({
  user: state.auth.user
});

const mapDispatchToProps = dispatch => {
  return {
    logout: () => {
      dispatch(Creators.logout());
    },
    load: () => {
      dispatch(Creators.loadRequest());
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MainPage);
