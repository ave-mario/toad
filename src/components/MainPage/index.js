import React, { Component } from 'react';
import { Title, Button } from '../../elements';
import store from '../../config/redux.store';

class MainPage extends Component {
  constructor(props) {
    super(props);
    this.handleLogout = this.handleLogout.bind(this);
  }

  // eslint-disable-next-line class-methods-use-this
  handleLogout() {
    store.dispatch({
      type: 'LOGOUT'
    });
  }

  render() {
    return (
      <>
        <Title>Main Page</Title>
        <Button type="button" onClick={this.handleLogout}>
          LOGOUT
        </Button>
      </>
    );
  }
}

export default MainPage;
