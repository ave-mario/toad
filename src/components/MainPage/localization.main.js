export default {
  ru: {
    labels: {
      title: 'Главная'
    }
  },
  en: {
    labels: {
      title: 'Main page'
    }
  }
};
