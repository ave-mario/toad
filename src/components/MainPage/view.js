import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';
import { Title, Button } from '../../elements';

class MainPage extends Component {
  constructor(props) {
    super(props);
    this.handleLogout = this.handleLogout.bind(this);
  }

  handleLogout() {
    const { logout } = this.props;
    logout();
  }

  render() {
    const { t } = this.props;
    return (
      <>
        <Title>Main Page</Title>
        <Button type="button" onClick={this.handleLogout}>
          {t('LogoutButton')}
        </Button>
      </>
    );
  }
}

MainPage.propTypes = {
  logout: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired
};

export default withTranslation('common')(MainPage);
