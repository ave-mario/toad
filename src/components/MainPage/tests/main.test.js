import React from 'react';
import { shallow } from 'enzyme';
import MainPage from '../index';

describe('Test case for testing main page', () => {
  test('Main page snapshot', () => {
    const component = shallow(<MainPage />);

    expect(component).toMatchSnapshot();
  });
});
