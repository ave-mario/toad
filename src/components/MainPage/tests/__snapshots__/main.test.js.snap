// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for testing main page Main page snapshot 1`] = `
<MainPage
  i18n={Object {}}
  logout={[Function]}
  t={[Function]}
  tReady={true}
/>
`;
